op-service/dial/active_rollup_provider.go

```diff
@@ -61,26 +67,22 @@ func newActiveL2RollupProvider(
 	}
 	cctx, cancel := context.WithTimeout(ctx, networkTimeout)
 	defer cancel()
-	_, err := rp.RollupClient(cctx)
+	ep := p.rollupUrls[p.rollupIndex]
+	rollupClient, err := p.rollupDialer(cctx, p.networkTimeout, p.log, ep)
 	if err != nil {
 		return nil, fmt.Errorf("dialing rollup client: %w", err)
 	}
-	return rp, nil
+	p.currentRollupClient = rollupClient
+	_, err = p.RollupClient(cctx)
+	if err != nil {
+		return nil, fmt.Errorf("setting provider rollup client: %w", err)
+	}
+	return p, nil
```

Comment:
If the sequencer with index 0 is offline at startup, the dialing will error and the ctor return that error. We should add a test for this case, seems we only have a test right now that an _inactive_ seq for index 0 still works, but not an _offline_ one. This was also the case when the nil-case dialing was still in `RollupClient()`.

There should be a way to already run the activity check loop at creation without duplicating the entire logic here. Maybe one way is to simply let `checkCurrentSequencer` return `false` if the current one is nil, and setting the field `currentRollupClient` to `-1` further up in the ctor when creating the `ActiveL2RollupProvider` so that the first index that is dialed is then 0.